app/services/address.ts

```diff
@@ -0,0 +1,43 @@
+import Service, { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import type FilterService from './filter-service';
+import type Store from '@ember-data/store';
+import type RouterService from '@ember/routing/router-service';
+import { task } from 'ember-concurrency';
+import { createLocationQuery } from 'frontend-burgernabije-besluitendatabank/utils/search/location-query';
+import type { Address } from 'frontend-burgernabije-besluitendatabank/controllers/agenda-items/types';
+
+export default class AddressService extends Service {
+  @service declare store: Store;
+  @service declare filterService: FilterService;
+  @service declare router: RouterService;
+
+  @tracked addressSuggestion: Address[] = [];
+  @tracked selectedAddress?: Address | null;
+
+  getSelectedAddress = task(async (fullAddress: string) => {
+    if (!fullAddress) {
+      return null;
+    }
+    this.addressSuggestion = await createLocationQuery(fullAddress);
+    return (this.selectedAddress = this.addressSuggestion.find(
+      (addressSuggestion) => {
+        if (addressSuggestion.fullAddress === fullAddress) {
+          this.selectedAddress = addressSuggestion;
+          return true;
+        }
+        return false;
+      },
+    ));
```

Comment:
Not all that clear to me 🤔 You set the result of the find to `this.selectedAddress` and in the find you are doing the same. Would just do the find  and than return `this.selectedAddress`.

Questioning why you added the tru and false here, maybe im missing something 🤔 

 ```suggestion
 this.selectedAddress = this.addressSuggestion.find((addressSuggestion) => addressSuggestion?.fullAddress === fullAddress);
 
 return this.selectedAddress
 ```